tox.ini

```diff
@@ -30,6 +30,7 @@ extras =
 deps =
     py-{devel}: git+https://github.com/ansible-community/molecule.git@main#egg=molecule[test]
 commands =
+    ansible-galaxy install -r requirements.yml
```

Comment:
@apatard 

Thanks for pointing it out.

The collection is required in https://github.com/ansible-community/molecule-plugins/blob/8867bd36ebdf64dff4a957cd2bbe6726708d7fc9/test/podman/test_func.py#L81

# The difference

This issue is introduced in this commit, https://github.com/ansible-community/molecule/commit/4dd129417ee73fb9111f0b05f68a3838530fb6b8

## Before the commit

The ansible is installed by package management tool(apt-get on ubuntu), and some collections will be installed automatically in `/usr/lib/python3/dist-packages`.
When pytest runs, it will use `/usr/bin/ansible` installed by apt-get, so the collection can be found. That's why it worked before.

## After the commit

The `ansible-core` is added in molecule's requirements, and `molecule-plugin` requires `molecule`, so tox will install Ansible in tox virtual enviroment, e.g. `/home/runner/work/molecule-plugins/molecule-plugins/.tox/py310/bin/ansible`.

## Why

Collections are installed in `/usr/lib/python3/dist-packages` before and after. 

Besides the default collection path like `~/.ansible/collections` and `/usr/share/ansible/collections`, Ansible will try to search collection from `sys.path`, but in non virtual environment, `'/usr/lib/python3/dist-packages'` is in `sys.path` while in virtual environment, it's not. The test is as below. The source code can be find in Reference.

### In non virtual environment

```shell

vagrant@vagrant:~$ /usr/bin/python3 -c "import sys; print (sys.path)"
['', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/home/vagrant/.local/lib/python3.10/site-packages', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages']
```

### In virtual environment

```shell
vagrant@vagrant:~$ python3 -m venv venv
vagrant@vagrant:~$ ./venv/bin/python3 -c "import sys; print (sys.path)"
['', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/home/vagrant/venv/lib/python3.10/site-packages']
```

# Solution

We can install collections explicitly in default collection path(`~/.ansible/collections`) where is respected in both non virtual environment and virtual environment.

Reference:

https://github.com/ansible/ansible/blob/0937cc486219663b6b6e6a178ef40798217864fa/lib/ansible/utils/collection_loader/_collection_finder.py#L243-L257


/cc @ssbarnea